cli/internal/cmd/iamcreategcp.go

```diff
@@ -0,0 +1,215 @@
+	dec, err := base64.StdEncoding.DecodeString(serviceAccountKeyBase64)
+	if err != nil {
+		return nil, err
+	}
+
+	out := make(map[string]string)
+	if err = json.Unmarshal(dec, &out); err != nil {
+		return nil, err
+	}
+	return out, nil
+}
+
+// parseGcpFlags parses and validates the flags of the iam create gcp command.
+func parseGcpFlags(cmd *cobra.Command) (gcpFlags, error) {
+	zone, err := cmd.Flags().GetString("zone")
+	if err != nil {
+		return gcpFlags{}, fmt.Errorf("parsing zone string: %w", err)
+	}
+	if !zoneRegex.MatchString(zone) {
+		return gcpFlags{}, fmt.Errorf("invalid zone string: %s", zone)
+	}
+
+	region, err := cmd.Flags().GetString("region")
+	if err != nil {
+		return gcpFlags{}, fmt.Errorf("parsing region string: %w", err)
+	}
+	if region == "" {
+		// Infer region from zone.
+		zoneParts := strings.Split(zone, "-")
+		region = fmt.Sprintf("%s-%s", zoneParts[0], zoneParts[1])
+	}
+	if !regionRegex.MatchString(region) {
+		return gcpFlags{}, fmt.Errorf("invalid region string: %s", region)
+	}
+
+	projectID, err := cmd.Flags().GetString("projectID")
+	if err != nil {
+		return gcpFlags{}, fmt.Errorf("parsing projectID string: %w", err)
+	}
+	// Source for regex: https://cloud.google.com/resource-manager/reference/rest/v1/projects.
+	if !projectIDRegex.MatchString(projectID) {
+		return gcpFlags{}, fmt.Errorf("invalid projectID string: %s", projectID)
+	}
+
+	serviceAccID, err := cmd.Flags().GetString("serviceAccountID")
+	if err != nil {
+		return gcpFlags{}, fmt.Errorf("parsing serviceAccountID string: %w", err)
+	}
+	if !serviceAccIDRegex.MatchString(serviceAccID) {
+		return gcpFlags{}, fmt.Errorf("invalid serviceAccountID string: %s", serviceAccID)
+	}
+
+	yesFlag, err := cmd.Flags().GetBool("yes")
+	if err != nil {
+		return gcpFlags{}, fmt.Errorf("parsing yes bool: %w", err)
+	}
+
+	return gcpFlags{
+		serviceAccountID: serviceAccID,
+		zone:             zone,
+		region:           region,
+		projectID:        projectID,
+		YesFlag:          yesFlag,
+	}, nil
+}
+
+// gcpFlags contains the parsed flags of the iam create gcp command.
+type gcpFlags struct {
+	serviceAccountID string
+	zone             string
+	region           string
+	projectID        string
+	YesFlag          bool
+}
+
+// TODO:
+// checkGcpAuth returns a boolean that indicates whether the user is authenticated with the gcloud SDK and an error.
+func checkGcpAuth() (bool, error) {
+	/* cmd := exec.Command("gcloud", "config", "list", "account", `--format "value(core.account)"`)
+	 if err := cmd.Run(); err != nil {
+		return false, fmt.Errorf(`could not execute "gcloud config list account": %w`, err)
+	}
+
+	email, err := cmd.Output()
+	if err != nil {
+		return false, fmt.Errorf("could not retrieve gcloud login: %w", err)
+	}
+
+	if !regexp.MustCompile(`^\w+([-+.']\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$`).MatchString(string(email)) {
+		return false, fmt.Errorf("invalid gcloud user")
+	}
+	*/
```

Comment:
Mh, currently we just let Terraform handle this and return its error message. 

If the error message looks clean enough, I would say let Terraform handle this. If not, then maybe a catch layer would be sufficient based on Terraform's error message?

Reply:
I think relying on the TF error messages is enough here. It was my initial idea to add custom checks, since I didn't know Terraform had such concise error messages. I think they are more than sufficient for the UX we try to achieve.  Therefore I removed the custom check now